cmake/CMakeLists.txt

```diff
@@ -244,6 +244,31 @@ option(onnxruntime_USE_AZURE "Build with azure inferencing support" OFF)
 option(onnxruntime_USE_LOCK_FREE_QUEUE "Build with lock-free task queue for threadpool." OFF)
 option(onnxruntime_FORCE_GENERIC_ALGORITHMS "Disable optimized arch-specific algorithms. Use only for testing and debugging generic algorithms." OFF)
 
+# DX for interop feature option
+option(onnxruntime_USE_DX_FOR_INTEROP "Build with the DX for Interop feature." OFF)
+
+if (onnxruntime_USE_DX_FOR_INTEROP)
+  add_compile_definitions(DX_FOR_INTEROP=1)
+else()
+  add_compile_definitions(DX_FOR_INTEROP=0)
```

Comment:
Does adding this make it harder to use as now I have to check the value and not just that it's defined?